docs/user_guide.rst

```diff
@@ -591,44 +592,79 @@ From within a real python, where ``SomePackage`` *is* installed globally, and is
-   contains requirements that specify a specific version.
+    SomePackage == 1.2.3
+    DependencyOfSomePackage == 4.5.6
 
-2. The installation is performed using :ref:`--no-deps <install_--no-deps>`.
-   This guarantees that only what is explicitly listed in the requirements file is
-   installed.
+Using :ref:`pip freeze` to generate the requirements file will ensure that not
+only the top-level dependencies are included but their sub-dependencies as
+well, and so on. Perform the installation using :ref:`--no-deps
+<install_--no-deps>` for an extra dose of insurance against installing
+anything not explicitly listed.
 
-3. None of the packages to be installed utilize the setup_requires keyword. See
-   :ref:`Controlling setup_requires<controlling-setup-requires>`.
+This strategy is easy to implement and works across OSes and architectures.
+However, it trusts PyPI, its CDN, and the network. It also relies on indices
+and find-links locations not allowing packages to change without a version
+increase. (PyPI does protect against this.)
 
-4. The installation is performed against an index or find-links location that is
-   guaranteed to *not* allow archives to be changed and updated without a
-   version increase.  While this is safe on PyPI, it may not be safe for other
-   indices. If you are working with an unsafe index, consider the `peep project
-   <https://pypi.python.org/pypi/peep>`_ which offers this feature on top of pip
-   using requirements file comments.
+Hash-checking Mode
+------------------
+
+Beyond pinning version numbers, you can add hashes against which to verify
+downloaded packages::
 
+    FooProject == 1.2 --hash:sha256=2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824
 
-.. _PyPI: http://pypi.python.org/pypi/
+This protects against compromises of PyPI, its CDN, the HTTPS certificate
+chain, and the network between you and the packages. It also guards
+against a package changing without its version number changing, on indexes
+that allow this. This approach is a good fit for automated server deployments.
 
+Hash-checking mode is a labor-saving alternative to running a private index
+server containing approved packages: it removes the need to upload packages,
+maintain ACLs, and keep an audit trail (which a VCS gives you on the
+requirements file for free). It can also substitute for a vendor library,
+providing easier upgrades and less VCS noise. It does not, of course,
+provide the availability benefits of a private index or a vendor library.
+
+For more, see :ref:`pip install\'s discussion of hash-checking mode <hash-checking mode>`.
 
 .. _`Installation Bundle`:
 
-Create an Installation Bundle with Compiled Dependencies
-********************************************************
+Installation Bundles
+--------------------
 
-You can create a simple bundle that contains all of the dependencies you wish
-to install using::
+An installation bundle is a single file containing all of a project's
```

Comment:
The wording here makes it sound like an "Installation Bundle" is an Official Thing in pip land, whereas it is not.